### PR TITLE
Move generated openAPI specs out of genericapiserver and make it configurable

### PR DESCRIPTION
**Pull request by mbohlool**
A follow up to #31468

Generated OpenAPI Spec does not belong to genericapiserver package. A new package "generated" created in hope of all generated codes goes into it in future. Openapi folder of that package contains generated definitions and generic API server will accept the definition map as a configuration parameter.

Reference: #13414

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32622)

### CODE REVIEW

File: pkg/generated/openapi/doc.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
```
Comment (smarterclayton) [inline pkg/generated/openapi/doc.go]: Does federation go in here?  Because we've tried to treat federation as distinct, and combining the openapi docs here but everything else be separate seems wrong. 

Package directory is fine, although @thockin almost certainly has opinions.

Comment (mbohlool) [inline pkg/generated/openapi/doc.go]: @nikhiljindal Can you comment on the federation types? I am not sure if we have a separate code generation for federation. If we do, you can separate them in a another PR (or I can do it if you teach me speak federation :) )

Comment (mbohlool) [inline pkg/generated/openapi/doc.go]: @smarterclayton anyway I think we should separate federation types in another PR. I think it is not related to this PR.

**Comment by mbohlool** [comment]
every PR that touches types will make this PR need rebase. I would not rebase it until I get LGTM (hopefully soon)

**Comment by smarterclayton** [comment]
Yeah hold up on rebases - I will try to look tomorrow if I can stop
finding p0's against 1.4

File: pkg/generated/openapi/doc.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/generated/openapi/doc.go]: That's fine.

**Comment by smarterclayton** [comment]
Changes LGTM, please rebase and I'll label.

**Comment by mbohlool** [comment]
Thanks @smarterclayton if changes look good, I can self label it after rebased and test passed.

**Comment by smarterclayton** [comment]
Yeah please do.

File: pkg/genericapiserver/config.go [hunk 2]
```diff
@@ -32,6 +32,7 @@ import (
 	"github.com/golang/glog"
 	"gopkg.in/natefinch/lumberjack.v2"
 
+	"k8s.io/kubernetes/cmd/libs/go2idl/openapi-gen/generators/common"
```
Comment (lavalamp) [inline pkg/genericapiserver/config.go]: Eventually I think the generator and generic apiserver will not live in the same repository, so I think this import path shouldn't go this direction. I'm OK with it temporarily (and possibly past me even suggested this?) but I think this definition package should actually be a subdirectory of genericapiserver.

So the import chain should eventually be [specific apiserver] -> generator -> genericapiserver.

**Comment by lavalamp** [comment]
LGTMing because this is a clear improvement, my comment can be fixed in a followup.
